File: api-gateway/src/test/java/com/jmsoftware/maf/apigateway/AuthorizationTests.java
package com.jmsoftware.maf.apigateway;

import lombok.extern.slf4j.Slf4j;
import lombok.val;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.util.AntPathMatcher;

/**
 * Description: AuthorizationTests, change description here.
 *
 * @author 钟俊（zhongjun）, email: zhongjun@toguide.cn, date: 1/13/2021 11:27 AM
 **/
@Slf4j
@SpringBootTest
public class AuthorizationTests {
    @Test
    void antPathMatcherTests() {
        val antPathMatcher = new AntPathMatcher();
        val urlMatched = antPathMatcher.match("/**", "/spring-boot-admin/common/app-info");
        log.info("urlMatched: {}", urlMatched);
        Assertions.assertTrue(urlMatched);
    }
}

File: api-gateway/api-gateway-bootstrap/src/test/java/com/jmsoftware/maf/apigateway/AuthorizationTests.kt
package com.jmsoftware.maf.apigateway

import com.jmsoftware.maf.common.util.logger
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import org.springframework.util.AntPathMatcher

/**
 * # AuthorizationTests
 *
 * Description: AuthorizationTests, change description here.
 *
 * @author Johnny Miller (锺俊), e-mail: johnnysviva@outlook.com, date: 4/16/22 10:15 PM
 */
internal class AuthorizationTests {
    companion object {
        private val log = logger()
    }

    @Test
    fun antPathMatcherTests() {
        val antPathMatcher = AntPathMatcher()
        val urlMatched1 = antPathMatcher.match("/spring-boot-admin/**", "/spring-boot-admin/common/app-info")
        log.info("urlMatched1: $urlMatched1")
        assertTrue(urlMatched1)
        val urlMatched2 = antPathMatcher.match("/**", "/spring-boot-admin/common/app-info")
        log.info("urlMatched2: $urlMatched2")
        assertTrue(urlMatched2)
    }
}
